blog/content/post/2020-06-30-trenchboot-grub-cbfs.md

```diff
@@ -0,0 +1,575 @@
+
+    ```
+    $ cp /nix/store/bqi0phr2rvwqlzfr3qj5117arxjhlbil-landing-zone-0.6.0/lz_header.bin /boot/lz_header
+    ```
+
+5. Reboot platform and boot to NixOS Secure Launch.
+
+6. Install `cbmem-tb` package manually.
+
+    ```
+    $ cd ~/nixpkgs
+    $ nix-build -A cbmem-tb
+    (...)
+    /nix/store/8y25mfqw0igqa5yfpvrks0nvr5wah5kn-cbmem-4.12
+    ```
+
+7. Copy `cbmem` utility to home directory.
+
+    ```
+    $ cd /nix/store/
+    $ ls | grep -i 'cbmem-4.12'
+    8y25mfqw0igqa5yfpvrks0nvr5wah5kn-cbmem-4.12
+    b4pds6l5fbkxdykkf2248c4mfnhf5ll5-cbmem-4.12.drv
+    plm3jyg54hjyiy3zldclx83k14i34lpq-cbmem-4.12.drv
+    pzifnqhgk0xarm3xrgkn07s9l572aw5p-cbmem-4.12.lock
+    ```
+
+    Search for entry with `cbmem-4.12` without any extension. In above case
+    `cbmem` is installed in
+    `/nix/store/8y25mfqw0igqa5yfpvrks0nvr5wah5kn-cbmem-4.12` directory.
+
+    ```
+    $ cp /nix/store/8y25mfqw0igqa5yfpvrks0nvr5wah5kn-cbmem-4.12/sbin/cbmem ~/
+    ```
+
+7. Go to home directory and use `cbmem` to read TPM event log.
+
+    ```
+    $ cd ~
+    $ ./cbmem -d
+    DRTM TPM2 log:
+        Specification: 2.00     Platform class: PC Client
+        No vendor information provided
+    DRTM TPM2 log entry 1:
+        PCR: 17
+        Event type: Unknown (0x600)
+        Digests:                 SHA1: 2400e5bdfbaa8cfc42eae13d9b742b89d0ba35b4
+                 SHA256: b65067767baf988b18e3a83410f90f055a4dcd59509b1ab6b17e18926ad8de82
+        Event data not provided
+    DRTM TPM2 log entry 2:
+        PCR: 17
+        Event type: Unknown (0x601)
+        Digests:                 SHA1: ecad3658a0cda535a8db50c207d726d2dac46509
+                 SHA256: f76f571cb78beddba64ceab81b679ab3328ec15e2c4ff49f9fff625c300dc5a1
+        Event data: Kernel
+    ```
+
+    > `-d` flag means 'print DRTM TPM log'. For more information about cbmem
+    usage, type `cbmem --help`.
+
+    Above logs are collected on platform with TPM2.0. Below there is example for
+    platform with TPM1.2.
+
+    ```
+    $ cd ~
+    $ ./cbmem -d
+    DRTM TCPA log:
+            Specification: 1.21     Platform class: PC Client
+            No vendor information provided
+    DRTM TCPA log entry 1:
+            PCR: 17
+            Event type: Unknown (0x600)
+            Digest: 2400e5bdfbaa8cfc42eae13d9b742b89d0ba35b4
+            Event data not provided
+    DRTM TCPA log entry 2:
+            PCR: 17
+            Event type: Unknown (0x601)
+            Digest: ecad3658a0cda535a8db50c207d726d2dac46509
+            Event data: Kernel
+    ```
+
+As you can see output for platform with TPM1.2 is slightly different than with
+TPM2.0. However, both are showing events and hashes which extend PCR 17. For
+now, there is lack of support for TPM event log in Linux kernel. Therefore,
+final values **are not the same** as those read with tpm_tools. We are going to
+add this feature in near future, so TPM event log will be complete and will be
+suitable for real-case use.
+
```

Comment:
@piotr-kleins I would add the verification of the 1st log entry (which should be LZ). For example use script to show the hash of LZ (calculated from binary not the extended hash) and compare it to the log entry.

Reply:
I was wondering how to verify them. That's good idea. I will add this

Reply:
I've added example for TPM2.0